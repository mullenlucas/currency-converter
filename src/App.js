import React from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import routes from './components/Routes';
import Nav from './components/Nav';
import Footer from './components/Footer';
import CurrenciesPage from './displays/currencyPage';

function App() {
  return (
    <div className="App">
      <Router className="App">
        <Nav title="Global Currency Converter" routes={routes} />
        <Routes>
          <Route path="/" element={<CurrenciesPage />} />
          <Route path="/converter" element={<p>Converter Page</p>} />
          <Route path="/date" element={<p>Date Page</p>} />
          <Route path="/*" element={<p>404 Page not found</p>} />
        </Routes>
        <Footer />
      </Router>
    </div>
  );
}

export default App;
