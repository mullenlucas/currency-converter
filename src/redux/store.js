// import { configureStore } from '@reduxjs/toolkit';

// const store = configureStore({
//   reducer: {
//     convert: convertReducer,
//   },
// });

// export default store;
