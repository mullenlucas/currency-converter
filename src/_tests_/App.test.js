import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import App from '../App';

describe('Top level App rendering test', () => {
  test('App snapshot rendering', () => {
    const tree = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
