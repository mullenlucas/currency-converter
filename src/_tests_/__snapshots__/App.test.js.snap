// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top level App rendering test App snapshot rendering... 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header>
          <a
            class="header-logo"
            href="/"
          >
            <img
              alt="logo currency converter"
              class="logo-img"
              src="currency-dice.png"
            />
            <h1>
              Global Currency Converter
            </h1>
          </a>
          <nav>
            <ul
              class="ul-links"
            >
              <li>
                <a
                  aria-current="page"
                  class="active"
                  href="/"
                  id="home"
                >
                  Currencies
                </a>
              </li>
            </ul>
          </nav>
        </header>
        <section
          class="body-section"
        >
          <div
            class="home-card"
          >
            <div
              class="hc-txt"
            >
              <h2>
                Real time currency converter to US dollars
              </h2>
              <h3>
                Discover and browse worldwide currencies and learn about them!
              </h3>
            </div>
            <div
              class="hc-img"
            >
              <img
                alt="illustration-exchange"
                class="illus-img"
                src="worldbills.png"
              />
            </div>
          </div>
          <div
            id="search"
          >
            <input
              placeholder="Search Currencies"
              type="search"
              value=""
            />
          </div>
          <div
            class="cards-container"
          />
        </section>
        <footer>
          <div
            class="footer-header"
          >
            <p>
              Lucas Mullen
            </p>
            <p>
              All rights reserved
            </p>
          </div>
          <div
            class="footer-footer"
          >
            <p>
              Go to
              <a
                href="https://github.com/mullenlucas/currency-converter"
              >
                 
                GitHub repository
                 
              </a>
              for more information and acknowledgements
            </p>
          </div>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header>
        <a
          class="header-logo"
          href="/"
        >
          <img
            alt="logo currency converter"
            class="logo-img"
            src="currency-dice.png"
          />
          <h1>
            Global Currency Converter
          </h1>
        </a>
        <nav>
          <ul
            class="ul-links"
          >
            <li>
              <a
                aria-current="page"
                class="active"
                href="/"
                id="home"
              >
                Currencies
              </a>
            </li>
          </ul>
        </nav>
      </header>
      <section
        class="body-section"
      >
        <div
          class="home-card"
        >
          <div
            class="hc-txt"
          >
            <h2>
              Real time currency converter to US dollars
            </h2>
            <h3>
              Discover and browse worldwide currencies and learn about them!
            </h3>
          </div>
          <div
            class="hc-img"
          >
            <img
              alt="illustration-exchange"
              class="illus-img"
              src="worldbills.png"
            />
          </div>
        </div>
        <div
          id="search"
        >
          <input
            placeholder="Search Currencies"
            type="search"
            value=""
          />
        </div>
        <div
          class="cards-container"
        />
      </section>
      <footer>
        <div
          class="footer-header"
        >
          <p>
            Lucas Mullen
          </p>
          <p>
            All rights reserved
          </p>
        </div>
        <div
          class="footer-footer"
        >
          <p>
            Go to
            <a
              href="https://github.com/mullenlucas/currency-converter"
            >
               
              GitHub repository
               
            </a>
            for more information and acknowledgements
          </p>
        </div>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
